hpcbench/benchmark/imb.py

```diff
@@ -16,18 +16,30 @@
 
 
 class IMBExtractor(MetricsExtractor):
+    def __init__(self):
+        self.with_all_data = False
+
     """Abstract class for IMB benchmark metrics extractor
     """
     @cached_property
     def metrics(self):
-        return dict(minb_lat=Metrics.Microsecond,
-                    minb_lat_bytes=Metrics.Byte,
-                    min_lat=Metrics.Microsecond,
-                    min_lat_bytes=Metrics.Byte,
-                    maxb_bw=Metrics.MegaBytesPerSecond,
-                    maxb_bw_bytes=Metrics.Byte,
-                    max_bw=Metrics.MegaBytesPerSecond,
-                    max_bw_bytes=Metrics.Byte)
+        common = dict(
+            minb_lat=Metrics.Microsecond,
+            minb_lat_bytes=Metrics.Byte,
+            min_lat=Metrics.Microsecond,
+            min_lat_bytes=Metrics.Byte,
+            maxb_bw=Metrics.MegaBytesPerSecond,
+            maxb_bw_bytes=Metrics.Byte,
+            max_bw=Metrics.MegaBytesPerSecond,
+            max_bw_bytes=Metrics.Byte
+        )
+        if self.with_all_data:
+            common.update(raw=list(dict(
```

Comment:
Hey @ohm314 
Here is an API change proposal to specify multi-valued and nested metrics:

* `list` means the metric is multi-valued.
* `dict` describes a nested metric.

Reply:
Excellent, @tristan0x ! This looks intuitive and makes sense to me.

Reply:
I took a look at your code PR and all changes make sense to me. I think this should do exactly what we need without adding too much complexity